include/aspect/utilities.h

```diff
@@ -604,6 +604,25 @@ namespace aspect
          */
         explicit AsciiDataLookup(const double scale_factor);
 
+        /**
+         * Replace the data stored in this class by the data given to this function.
+         *
+         * @note This is a manual way to fill the data. Consider load_file() if your data
+         * is stored in a file.
+         *
+         * The data consists of @p n_components (implicitly given by the size of @p column_names)
+         * specified at @p n_points_per_direction in each of the dim coordinate directions.
+         *
+         * The data in @p raw_data consists of dim+n_components number of tables, where the
+         * first dim tables are the coordinate values (raw_data[0] is the table of x
+         * coordinates, raw_data[1] is the table of y coordinates, etc..) and the remaining
+         * columns are the data columns.
+         */
+        void reinit(const TableIndices<dim> &n_points_per_direction,
```

Comment:
do you actually need this parameter? I'd think you can get that as the size of one of the data tables.